api/client.go

```diff
@@ -550,19 +676,24 @@ func (c *Client) Token() string {
 
 // SetToken sets the token directly. This won't perform any auth
 // verification, it simply sets the token properly for future requests.
+// setting the token to "" will return to polling a file sink if one is available
 func (c *Client) SetToken(v string) {
-	c.modifyLock.Lock()
-	defer c.modifyLock.Unlock()
-
-	c.token = v
+	if v != "" {
+		c.modifyLock.Lock()
+		defer c.modifyLock.Unlock()
+		c.useFileSinkForToken = false
+		c.token = v
+	} else {
+		c.modifyLock.Lock()
```

Comment:
Shouldn't we set c.token = v in both branches?

Reply:
yes... otherwise it is reverting to polling but not "clearing" the token.